packages/local/src/index.ts

```diff
@@ -0,0 +1,161 @@
+import { Context, Logger, Random, Schema } from 'koishi'
+import { ImageSource } from 'koishi-plugin-booru'
+import { LocalStorage } from './types'
+import { createHash } from 'node:crypto'
+import { PathOrFileDescriptor, existsSync, readFileSync, readdirSync, statSync, writeFile } from 'node:fs'
+import { basename, extname, isAbsolute, join, resolve, sep } from 'node:path'
+import { pathToFileURL } from 'node:url'
+
+class LocalImageSource extends ImageSource<LocalImageSource.Config> {
+  languages = ['en', 'zh-CN', 'ja']
+  source = 'local'
+  static usage = `
+## 使用说明
+
+插件启动时会扫描文件夹并建立数据标记，这将会耗费较长的时间，请耐心等待。
+`
+  private imageMap: LocalStorage.Type[] = []
+  private logger: Logger
+
+  constructor(ctx: Context, config: LocalImageSource.Config) {
+    super(ctx, config)
+    this.logger = ctx.logger('booru-local')
+
+    ctx.on('ready', () => {
+      if (config.endpoint.length <= 0) {
+        this.logger.warn('no folder yet')
+        return
+      } else {
+        this.init(config.endpoint)
+      }
+    })
+  }
+
+  private hash(path: PathOrFileDescriptor, file: boolean = false) {
+    return createHash('md5').update(file ? readFileSync(path) : path.toString()).digest('hex')
+  }
+
+  async init(paths: string[]) {
+    const _name = 'booru-map.json'
+    paths.forEach(path => {
```

Comment:
```suggestion
    paths.forEach((path) => {
```

We prefer brackets in the arrow function parameters even if there is only one.